app/models/geocodable.rb

```diff
@@ -21,8 +21,12 @@ def coordinates
     [latitude, longitude]
   end
 
-  def update_coordinates!(coordinates)
+  def coordinates=(coordinates)
     self.latitude, self.longitude = coordinates
-    update_columns(latitude: latitude, longitude: longitude)
+  end
+
+  def update_coordinates!(coordinates)
+    self.coordinates = coordinates
+    update_columns(latitude: latitude, longitude: longitude) if persisted?
```

Comment:
Thanks as discussed. `if persisted?` should NOT be here, the need for it was replaced by `def coordinates=` but I clearly forgot to go back and take it out :( my bad.